src/modules/navigator/mission.cpp

```diff
@@ -953,10 +951,10 @@ Mission::altitude_sp_foh_update()
 		* The setpoint is set linearly and such that the system reaches the current altitude at the acceptance
 		* radius around the current waypoint
 		**/
-		float delta_alt = (get_absolute_altitude_for_item(_mission_item) - _mission_item_previous_alt);
+		float delta_alt = (get_absolute_altitude_for_item(_mission_item) - pos_sp_triplet->previous.alt);
```

Comment:
Because there never was a position setpoint yet. It would be possible to just set it to the current location when mission is activate. Probably makes sense anyway even if it prev is already set. Then you don't have to care what happened before.